src/Bicep.Core/Workspaces/SourceFileGroupingBuilder.cs

```diff
@@ -264,5 +288,169 @@ private void PopulateRecursive(BicepSourceFile file, IFeatureProviderFactory fea
+                {
+                    return new(uriResolutionError.ErrorBuilder);
+                }
+                return new(new ResourceTypesProviderDescriptor(
+                               inlinedSpecification.NamespaceIdentifier,
+                               file.FileUri,
+                               inlinedSpecification.Version,
+                               providerDeclarationSyntax.Alias?.IdentifierName,
+                               artifactReference,
+                               typesDataUri));
+            }
+
+            // The provider is configuration managed, fetch the provider source & version from the configuration
+            var configLookupValue = providerDeclarationSyntax.Specification.NamespaceIdentifier;
+
+            // Special case the "sys" provider for backwards compatibility
+            if (configLookupValue == SystemNamespaceType.BuiltInName)
+            {
+                return new(new ResourceTypesProviderDescriptor(SystemNamespaceType.BuiltInName, file.FileUri));
+            }
+
+            if (!config.ProvidersConfig.TryGetProviderSource(configLookupValue).IsSuccess(out var providerEntry, out var errorBuilder))
+            {
+                return new(errorBuilder);
+            }
+
+            //handle built-in providers
+            if (providerEntry.BuiltIn)
+            {
+                return new(new ResourceTypesProviderDescriptor(configLookupValue, file.FileUri));
+            }
+
+            var unreachableException = new UnreachableException("artifact path is validated during artifact creation so Source & Version must cannot be null");
+            var providerNamespaceIdentifier = providerEntry.Source?.Split('/')[^1] ?? throw unreachableException;
+
+            if (!uriResult.IsSuccess(out var uri, out var uriResoulutionError))
+            {
+                return new(uriResoulutionError.ErrorBuilder);
+            }
+
+            return new(new ResourceTypesProviderDescriptor(
+                providerNamespaceIdentifier,
+                file.FileUri,
+                providerEntry.Version ?? throw unreachableException,
+                providerDeclarationSyntax.Alias?.IdentifierName ?? configLookupValue,
+                artifactReference,
+                uri));
+        }
+
+        private void ProcessAllImplicitProviderDeclarations(BicepSourceFile file, ProviderDescriptorBundleBuilder providerBundleBuilder, RootConfiguration config)
+        {
+            foreach (var providerName in config.ImplicitProvidersConfig.GetImplicitProviderNames())
+            {
+                try
+                {
+                    ProcessSingleImplicitProviderDeclaration(providerName, providerBundleBuilder, file, config);
+                }
+                catch (Exception ex)
+                {
+                    this.providerDescriptorBundleBuilderBySourceFile[file].AddImplicitProvider(new(x => x.ArtifactRestoreFailedWithMessage(providerName, ex.Message)));
```

Comment:
Why are we expecting an exception here? Can we be any more specific about the type of exception we're expecting?

Reply:
removed the try/catch block